components/user_settings/security/user_access_token_section/user_access_token_section.jsx

```diff
@@ -0,0 +1,576 @@
+                        values={{
+                            description: token.description,
+                        }}
+                    />
+                </div>
+            ),
+            confirmButton: (
+                <FormattedMessage
+                    id='user.settings.tokens.confirmDeleteButton'
+                    defaultMessage='Yes, Delete'
+                />
+            ),
+            confirmComplete: () => {
+                this.revokeToken(tokenId);
+                trackEvent('settings', 'revoke_user_access_token');
+            },
+        });
+    }
+
+    revokeToken = async (tokenId) => {
+        const {error} = await this.props.actions.revokeUserAccessToken(tokenId);
+        if (error) {
+            this.setState({serverError: error.message});
+        }
+        this.handleCancelConfirm();
+    }
+
+    activateToken = async (tokenId) => {
+        const {error} = await this.props.actions.enableUserAccessToken(tokenId);
+        if (error) {
+            this.setState({serverError: error.message});
+        } else {
+            trackEvent('settings', 'activate_user_access_token');
+        }
+    }
+
+    deactivateToken = async (tokenId) => {
+        const {error} = await this.props.actions.disableUserAccessToken(tokenId);
+        if (error) {
+            this.setState({serverError: error.message});
+        } else {
+            trackEvent('settings', 'deactivate_user_access_token');
+        }
+    }
+
+    render() {
+        let tokenListClass = '';
+
+        if (!this.props.active) {
+            const describe = Utils.localizeMessage('user.settings.tokens.clickToEdit', "Click 'Edit' to manage your personal access tokens");
+
+            return (
+                <SettingItemMin
+                    title={Utils.localizeMessage('user.settings.tokens.title', 'Personal Access Tokens')}
+                    describe={describe}
+                    section={SECTION_TOKENS}
+                    updateSection={this.props.updateSection}
+                />
+            );
+        }
+
+        const tokenList = [];
+        Object.values(this.props.userAccessTokens).forEach((token) => {
+            if (this.state.newToken && this.state.newToken.id === token.id) {
+                return;
+            }
+
+            let activeLink;
+            let activeStatus;
+
+            if (token.is_active) {
+                activeLink = (
+                    <a
+                        name={token.id + '_deactivate'}
+                        href='#'
+                        onClick={(e) => {
+                            e.preventDefault();
+                            this.deactivateToken(token.id);
+                        }}
+                    >
+                        <FormattedMessage
+                            id='user.settings.tokens.deactivate'
+                            defaultMessage='Deactivate'
+                        />
+                    </a>);
```

Comment:
nit: This is also probably from the old code, but the parentheses ends on a different line than line 302 would suggest.